AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -202,6 +204,17 @@ - (instancetype)initWithFrame:(CGRect)frame collectionViewLayout:(UICollectionVi
   // and should not trigger a relayout.
   _ignoreMaxSizeChange = CGSizeEqualToSize(_maxSizeForNodesConstrainedSize, CGSizeZero);
   
+  // Register the default layout inspector delegate for flow layouts, custom layouts
+  // will need to roll their own ASCollectionViewLayoutInspecting implementation.
+  if ([layout asdk_isFlowLayout]) {
+    _flowLayoutInspector = [[ASCollectionViewFlowLayoutInspector alloc] init];
+    _flowLayoutInspector.layout = (UICollectionViewFlowLayout *)layout;
+    _flowLayoutInspector.collectionView = self;
+    _layoutDelegate = _flowLayoutInspector;
+  }
+  
+  _registeredSupplementaryKinds = [NSMutableArray array];
```

Comment:
Simple optimization, small impact but easy — lazy initialize this on first call of register.